workflow/validate/validate.go

```diff
@@ -351,6 +352,15 @@ func (ctx *templateValidationCtx) validateTemplate(tmpl *wfv1.Template, tmplCtx
 		return err
 	}
 
+	if tmpl.IsLeaf() {
+		// inline template may not have a name
+		if len(tmpl.Name) != 0 {
```

Comment:
This would be a breaking change for some users. Perhaps we should lowercase them before validation?

Reply:
Thanks for the reminder, the corresponding compatibility fixes have been made here.

But the price of doing this is that we will accept that the user's template name on the Leaf node contains uppercase. Under the naming method of Pod v1, a Pod whose name contains uppercase characters will be created. This is illegal in k8s.